drivers/gpio/gpio_pcf857x.c

```diff
@@ -365,33 +383,37 @@ static int pcf8574_init(const struct device *dev)
 	return 0;
 }
 
-/** Realizes the functions of gpio.h for pcf8574*/
-static const struct gpio_driver_api pcf8574_drv_api = {
-	.pin_configure = pcf8574_pin_configure,
-	.port_get_raw = pcf8574_port_get_raw,
-	.port_set_masked_raw = pcf8574_port_set_masked_raw,
-	.port_set_bits_raw = pcf8574_port_set_bits_raw,
-	.port_clear_bits_raw = pcf8574_port_clear_bits_raw,
-	.port_toggle_bits = pcf8574_port_toggle_bits,
-	.pin_interrupt_configure = pcf8574_pin_interrupt_configure,
-	.manage_callback = pcf8574_manage_callback,
+/** Realizes the functions of gpio.h for pcf857x*/
+static const struct gpio_driver_api pcf857x_drv_api = {
+	.pin_configure = pcf857x_pin_configure,
+	.port_get_raw = pcf857x_port_get_raw,
+	.port_set_masked_raw = pcf857x_port_set_masked_raw,
+	.port_set_bits_raw = pcf857x_port_set_bits_raw,
+	.port_clear_bits_raw = pcf857x_port_clear_bits_raw,
+	.port_toggle_bits = pcf857x_port_toggle_bits,
+	.pin_interrupt_configure = pcf857x_pin_interrupt_configure,
+	.manage_callback = pcf857x_manage_callback,
 };
 
-#define GPIO_PCF8574_INST(idx)                                                                     \
-	static const struct pcf8574_drv_cfg pcf8574_cfg##idx = {                                   \
+#define GPIO_PCF857X_INST(idx)                                                                     \
+	static const struct pcf857x_drv_cfg pcf857x_cfg##idx = {                                   \
 		.common =                                                                          \
 			{                                                                          \
 				.port_pin_mask = GPIO_PORT_PIN_MASK_FROM_DT_INST(idx),             \
 			},                                                                         \
 		.gpio_int = GPIO_DT_SPEC_INST_GET_OR(idx, int_gpios, {0}),                         \
 		.i2c = I2C_DT_SPEC_INST_GET(idx),                                                  \
 	};                                                                                         \
```

Comment:
You should be able to calculate the number of bytes at compile time, using something like:

```suggestion
		.num_bytes = DT_INST_ENUM_IDX(idx, ngpios) + 1,	
	};                                                                                         \
```